### DISCUSSION TITLE
Default value for variable that will hold a MX sym

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3478] [Q&A]
Dear developers,

I am assiging default values for certain varibles that will take MX type as empty MX matrices inside a class. Later I check if the user has entered a value for these variables and accordingly write the logic for the program.

In my case there is time variable in the code which can be constant or an optimization variable. If nothing is specified, it takes the default value and becomes a part of the decision vector. Otherwise, it is simply a constant value of type MX.

Present I check if the time variable is empty or not and add it to optimization problem using is_empty() method. is_constant() is another method I wanted to try out but it returns true even for empty matrix and does not help in my case.

1. what is a good default value for a variable of type MX, SX and DM?
2.  some methods I can use to distinguish between default and user suplied?

```
import casadi as cs
y1=cs.SX()
print(y1)
print(y1.is_integer())
```

**Comment by jgillis** [comment]
`is_empty()` may be what you are looking for.

**Comment by sandeep026** [comment]
Yes. That's what I had used
